src/main/java/net/datafaker/transformations/JsonTransformer.java

```diff
@@ -7,149 +7,185 @@
+        StringBuilder sb = new StringBuilder();
+        for (int i = 0; i < limit; i++) {
+            sb.append(apply(null, schema, i));
+            if (i < limit - 1) {
+                sb.append(",").append(LINE_SEPARATOR);
+            }
+        }
+        char[] delimitedBy = delimitedBy(this.formatAsArray);
+        return limit > 1 ? delimitedBy[0] + LINE_SEPARATOR + sb + LINE_SEPARATOR + delimitedBy[1] : sb.toString();
+    }
+
+    private static void value2String(Object value, StringBuilder sb) {
+        if (value == null) {
+            sb.append("null");
+        } else if (value instanceof Integer
+            || value instanceof Long
+            || value instanceof Short
+            || value instanceof BigInteger
+            || value instanceof Boolean
+            || (value instanceof Double
             && BigDecimal.valueOf((Double) value).remainder(BigDecimal.ONE).doubleValue() == 0)
-        || (value instanceof BigDecimal
+            || (value instanceof BigDecimal
             && ((BigDecimal) value).remainder(BigDecimal.ONE).doubleValue() == 0)) {
-      sb.append(value);
-    } else if (value instanceof Collection) {
-      sb.append(generate((Collection) value));
-    } else if (value.getClass().isArray()) {
-      sb.append(generate(Arrays.asList((Object[]) value)));
-    } else {
-      String val = String.valueOf(value);
-      boolean toWrap = !val.startsWith("#{json");
-      if (toWrap) {
-        sb.append("\"");
-      }
-      for (char c : String.valueOf(value).toCharArray()) {
-        sb.append(ESCAPING_MAP.getOrDefault(c, c + ""));
-      }
-      if (toWrap) {
-        sb.append("\"");
-      }
+            sb.append(value);
+        } else if (value instanceof Collection) {
+            sb.append(generate((Collection) value));
+        } else if (value.getClass().isArray()) {
+            sb.append(generate(Arrays.asList((Object[]) value)));
+        } else {
+            String val = String.valueOf(value);
+            boolean toWrap = !val.startsWith("#{json");
+            if (toWrap) {
+                sb.append("\"");
+            }
+            for (char c : String.valueOf(value).toCharArray()) {
+                sb.append(ESCAPING_MAP.getOrDefault(c, c + ""));
+            }
+            if (toWrap) {
+                sb.append("\"");
+            }
+        }
+    }
+
+    private static String generate(Collection<Object> collection) {
+        StringBuilder sb = new StringBuilder();
+        sb.append("[");
+        int i = 0;
+        for (Object value : collection) {
+            if (i > 0) {
+                sb.append(", ");
+            }
+            i++;
+            value2String(value, sb);
+        }
+        sb.append("]");
+        return sb.toString();
+    }
+
+
+    private static Map<Character, String> createEscapeMap() {
+        final Map<Character, String> map = new HashMap<>();
+        map.put('\\', "\\\\");
+        map.put('\"', "\\\"");
+        map.put('\b', "\\b");
```

Comment:
Why do we need to touch these lines within this PR?

Reply:
editorconfig says 4 indent for java - this class was 2.

Reply:
it should be in a separate commit